Make nav editor links content-width instead of expanding

vcanales opened the pull request:
## Description
Fixes https://github.com/WordPress/gutenberg/issues/34036
Make sure Navigation Items don't expand their width within the Navigation Editor, when longer items are present.

## How has this been tested?
1. Go to Nav Editor, create items of different lengths.
2. Shorter items should stick to content-size instead of expanding.

## Screenshots <!-- if applicable -->

![Kapture 2021-09-08 at 12 12 03](https://user-images.githubusercontent.com/1157901/132536323-15875fb6-1c2e-45c8-9644-67bae9064ee9.gif)

## Types of changes
Bug fix.

vcanales committed:
Make nav editor links content-width

gwwar commented:
I think the proposed changes here looks reasonable, but it looks like this may affect the vertical distance between each item in the navigation editor.

https://user-images.githubusercontent.com/1270189/132548272-d91d72bc-eeea-4cce-974b-2d4d840ed98d.mp4

Is this also an issue in the published page? Should we also add similar behavior to the vertical navigation block variation in other editors?

<img width="1007" alt="CleanShot 2021-09-08 at 09 30 58@2x" src="https://user-images.githubusercontent.com/1270189/132548639-eeeb2838-58d2-4780-b996-552f2c037804.png">

jasmussen commented:
Enabling flex might allow the new `gap` rules to take. Whether that's something desired for the editor screen is likely a question for @javierarce or @getdave. Thanks for looking!

javierarce commented:
I think the increase of vertical distance is acceptable (it also makes it easier to insert elements between items).

The only problem I see with this change (and this is something I hadn't thought of until now), is that the inserter doesn't have a consistent width, since it depends on the item above:

<img width="470" alt="Screenshot 2021-09-10 at 11 08 00" src="https://user-images.githubusercontent.com/4933/132830546-958384d4-e994-48de-b825-b9257e89afd8.png">

<img width="504" alt="Screenshot 2021-09-10 at 11 08 08" src="https://user-images.githubusercontent.com/4933/132830541-89e9764f-d90a-4bc5-881a-6155c86ac97b.png">

I don't think is a stopper, but it's definitely something we could address here https://github.com/WordPress/gutenberg/issues/34035

getdave reviewed (commented):
Just noting that if we ever want to set a background colour on nav items then this change might make things look a little odd.

talldan commented:
I've just rebased this locally against `trunk`, as there were quite a few styling fixes in #34281, which was just merged.

The main problem I'm seeing with the increased spacing is that it only applies to the first level of items:
<img width="496" alt="Screenshot 2021-09-13 at 12 28 59 pm" src="https://user-images.githubusercontent.com/677833/133024166-a7a936d6-e72b-443e-88e4-c3e43759f552.png">

In general I'm thinking it might be good to have consistent spacing at all levels and I prefer the spacing in `trunk` because otherwise menus end up very big. Especially if a way to expand all submenus is introduced later on.

The second issue I'm seeing is the trailing appender at root level is now centered with the longest item:
<img width="433" alt="Screenshot 2021-09-13 at 12 32 01 pm" src="https://user-images.githubusercontent.com/677833/133024357-96b0a6c4-ad94-4332-a094-a65c9bc7b97b.png">

vcanales commented:
The spacing issue is due to vertical margin collapse not being a feature of flex layouts, so I'm currently looking into either working around this limitation by halving margins where appropriate, or if unable, finding a way to achieve content-width elements without a flex layout.

talldan commented:
@vcanales I know @jasmussen changed some styles to flex and fixed some similar spacing issues to the ones you're seeing, so it'd be interesting to see how this looks after a rebase.

Commit:
remove gap, align items on submenu container

vcanales commented:
Addressed the different items spacing problems; one of the issues was a `gap` being set for flex containers elsewhere in the codebase.

|This change|trunk|
|--|--|
|![Kapture 2021-10-08 at 12 25 55](https://user-images.githubusercontent.com/1157901/136583367-3c892d29-414f-475b-bb50-b36060cc0e35.gif)|![Kapture 2021-10-08 at 12 32 36](https://user-images.githubusercontent.com/1157901/136584179-107fc30d-d9b2-4e55-a6b4-132b947103a5.gif)|

> The only problem I see with this change (and this is something I hadn't thought of until now), is that the inserter doesn't have a consistent width, since it depends on the item above:

Worth mentioning that the issue where the squiggly line on empty links was solved elsewhere, as can be seen on the comparison above, making this PR only about having items be content-width. With that in mind, does this change perhaps create more problems than it solves? @javierarce 

If we move forward, this reverts one of the changes made by @jasmussen in https://github.com/WordPress/gutenberg/pull/35234, but doesn't seem to make the regression come back; I'd still appreciate 👀 on it!

> @vcanales I know @jasmussen changed some styles to flex and fixed some similar spacing issues to the ones you're seeing, so it'd be interesting to see how this looks after a rebase.

@talldan It still required the use of `flex`, but the issue with the squiggly line extending to the width of the element was solved without the need for this PR.

jasmussen reviewed (approved):
This tests well for me. Trunk and this branch look mostly the same — trunk:

![trunk](https://user-images.githubusercontent.com/1204802/136754729-f5769d16-1b87-4bbd-b6a2-f73661a402be.gif)

This branch:
![this branch](https://user-images.githubusercontent.com/1204802/136754753-69394a30-a220-4197-923a-500c0c4cbee2.gif)

The "clicking outside selects the item" is the same in both, I just tinkered more with it in the second recording.

The wavy underline was also fixed separately by a span wrapper. Nevertheless this is a nice little code cleanup. It reduces the height of items a little bit, but looks good to me.

Mamaduka commented:
Let's merge or close this PR. The navigation screen is no longer actively developed.

Related #43620.

vcanales commented:
Closing, since code will probably be removed.
